core/txpool/txpool.go

```diff
@@ -750,7 +749,7 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (replaced bool, err e
 			// Add all transactions back to the priced queue
 			if replacesPending {
 				for _, dropTx := range drop {
-					heap.Push(&pool.priced.urgent, dropTx)
+					pool.priced.Put(dropTx, local || pool.locals.containsTx(dropTx))
```

Comment:
These transactions are popped from the global price list, which are assumed non-local. You can simply pass the `local=False` here.

Reply:
> These transactions are popped from the global price list, which are assumed non-local. You can simply pass the `local=False` here.

RIght